R/api_url.R

```diff
@@ -124,7 +126,12 @@ api_url <- function(
     url <- paste0(
       endpoint_base_version,
       "publications?",
-      api_url_pages(page_size = page_size, page = page)
+      api_url_pages(page_size = page_size, page = page),
```

Comment:
Should we handle the conversion of spaces?

![image](https://github.com/user-attachments/assets/4202a77c-597f-4e20-8267-f895daf79437)

Reply:
I've converted " " to "-", which creates an OR query on the search terms. I did test "+" and "%20" as what sounds like the standard terms for just substituting for spaces in GET search queries, but they result in OR as well.
Have added a bunch of extra warning / error logic to help the user understand if something unexpected happens.

Reply:
Works well!